### PR TITLE
[BEAM-409] Fixing incorrect use of Math.ceil in ApproximateQuantiles

### PR DESCRIPTION
**Pull request by youngoli**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

Fixed the FindBug issue by avoiding an incorrect integer division. I based this on an old unmerged pull request for this bug: https://github.com/apache/beam/pull/853

That PR also included a new unit test for ApproximateQuantiles.java, wasn't sure if I should also include that unit test.

### CODE REVIEW

**Comment by swegner** [comment]
Thanks for this contribution. Please also include a unit test to verify. The unmerged PR #853 had a couple new tests, `testEfficiency()` and `testCorrectness()`, and just needed some slight changes to use JUnit `@Parameters`. See: https://github.com/apache/beam/pull/853/files#r75763207

**Comment by youngoli** [comment]
Thanks swegner. I added in the unit tests you made for PR #853 and added the `@Parameters` change.

**Comment by swegner** [comment]
retest this please

**Comment by swegner** [comment]
retest this please

**Comment by swegner** [comment]
LGTM. Adding @tgroh as Beam committer to merge.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ApproximateQuantilesTest.java [hunk 1]
```diff
@@ -45,266 +49,360 @@
...
+     * Verify that buffers are correct according to the two constraint equations.
+     */
+    @Test
+    public void testCorrectness() {
+      int b = combineFn.getNumBuffers();
+      int k = combineFn.getBufferSize();
+      long n = this.maxInputSize;
+
+      assertTrue("(b-2)2^(b-2) + 1/2 <= eN", (b - 2) * (1 << (b - 2)) + 0.5 <= this.epsilon * n);
+      assertTrue("k2^(b-1) >= N", Math.pow(k * 2, b - 1) >= n);
```
Comment (tgroh) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ApproximateQuantilesTest.java]: `assertThat("what", actual, Matchers.lessThanOrEqualTo(expected))` and the same with the greater than or equal to.

**Comment by youngoli** [comment]
retest this please

**Comment by youngoli** [comment]
retest this please

**Comment by youngoli** [comment]
retest this please

**Pull request closed without merging**